adapters/risc0/src/metrics.rs

```diff
@@ -17,17 +18,22 @@ pub fn add_value(metric: String, value: u64) {
         .or_insert((value, 1));
 }
 
-pub fn deserialize_custom(serialized: &[u8]) -> (String, u64) {
-    let null_pos = serialized.iter().position(|&b| b == 0).unwrap();
+pub fn deserialize_custom(serialized: Bytes) -> Result<(String, u64), anyhow::Error> {
+    let null_pos = serialized
+        .iter()
+        .position(|&b| b == 0)
+        .ok_or(anyhow::anyhow!(
+            "Could not find separator in provided bytes"
+        ))?;
```

Comment:
nit: use `.context` (or `.with_context` and print bytes)

Reply:
Good idea! Done 👍